lib/bolt/transport/docker.rb

```diff
@@ -112,7 +112,7 @@ def run_task(target, task, arguments, _options = {})
             end
 
             stdout, stderr, exitcode = conn.execute(remote_task_path, execute_options)
-            Bolt::Result.for_task(target, stdout.join, stderr.join, exitcode)
+            Bolt::Result.for_task(target, stdout.join, stderr.join.force_encoding("UTF-8"), exitcode)
```

Comment:
I added this to get tests passing because Non-utf characters from sterr were causing https://github.com/puppetlabs/bolt/blob/e4fe443c90e151ed087cbbb23bbc76a401cba12b/lib/bolt/result.rb#L119-L121 to stack trace. Any suggestions for where to make sure these characters are serializeable?

Reply:
What was the scenario in which you were seeing non-utf characters? I'm surprised this worked, because generally all `force_encoding` does is cause the string to _claim_ it's encoded in UTF-8. I would expect a `valid_encoding?` check to fail after doing that, because the bytes are still not actually valid UTF-8.

Reply:
Running interpreter.py against `ubuntu_node` defined in bolt/spec. You can reproduce with:

`test.py`
```
#!/foo/bar
print('nope')
```
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt$ bolt task run test::test -n ubuntu_node --transport docker
Started on ubuntu_node...
Failed on ubuntu_node:
<Encoding::UndefinedConversionError> "\xE2" from ASCII-8BIT to UTF-8
	/home/cas/working_dir/bolt/lib/bolt/result.rb:120:in `encode'
	/home/cas/working_dir/bolt/lib/bolt/result.rb:120:in `to_json'
	/home/cas/working_dir/bolt/lib/bolt/result.rb:120:in `to_json'
	/home/cas/working_dir/bolt/lib/bolt/executor.rb:205:in `with_node_logging'
	/home/cas/working_dir/bolt/lib/bolt/executor.rb:252:in `block (2 levels) in run_task'
	/home/cas/working_dir/bolt/lib/bolt/executor.rb:84:in `block (3 levels) in queue_execute'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
	/home/cas/working_dir/bolt/.bundle/ruby/2.5.0/gems/logging-2.2.2/lib/logging/diagnostic_context.rb:474:in `block in create_with_logging_context'
  The task failed with exit code 127:
  env: ‘/tmp/9dc6641e-c583-4f5e-a4ba-13bd4d1c423b/test/tasks/test.ps1’: No such file or directory
 
  {
  }
Failed on 1 node: ubuntu_node
Ran on 1 node in 1.06 seconds
```